### DISCUSSION TITLE
Problems with resourcepack creation in Minicraft 2.2.1

### DISCUSSION THREAD

**Discussion by Sernimator** [MinicraftPlus/minicraft-plus-revived#705] [Q&A] [answered]
I wanted to create a resourcepack with changes the font, I tried to do everything like in the guide, resourcepack has loaded, but didn't work. Then I tried to arrange them like in the jar file "Resourcepack_name.zip\assets\textures\gui", but textures still didn't work. Pack.json is in the folder, but it may be incorrectly filled in.

Is it possible to create a resourcepacks for 2.2.1? And if so, where can I find examples of resourcepacks?

I'm sorry if I seemed rude or illiterate, it's hard for me to write in English without a translator

**Comment by BenCheung0422** [comment]
The guides may be a bit outdated. Which one did you refer to?

**Reply by Sernimator** [reply]
I've tried both the wiki guide and the github guide

![Screenshot_20241023_062536](https://github.com/user-attachments/assets/6bc4bd55-6824-47fe-8c1f-e9c1dcdf9794)
![Screenshot_20241023_062337](https://github.com/user-attachments/assets/7023ed52-b45e-4cbf-80d3-db1c32600822)

**Reply by BenCheung0422** [reply]
Right, it seems to be an unmentioned part. Fonts are in `assets/textures/gui/font.png`. When you created a zip file, make sure that it is the path inside the zip file.

**Reply by Sernimator** [reply]
The problem is that it is located in assets/textures/gui/font.png, but it still doesn't work
![Screenshot_20241023_071303](https://github.com/user-attachments/assets/51cc5ada-283b-4f07-9e94-1f829f77a495)
I tried to replace the texts and titles, they worked correctly. Could this have happened because I was using an inappropriate graphics editor?

**Reply by BenCheung0422** [reply]
If your layout is the same as the one as in the game's one, it should work. What texts specifically you meant? I think using different editor does not affect in this way.

**Reply by Sernimator** [reply]
I changed Antidous text, from assets/books

**Reply by BenCheung0422** [reply]
Is your `pack_format` in `pack.json` set? If your `pack.json` contains enough data, and your `font.png` is drawn correctly, it should be able to load your pack.

**Reply by Sernimator** [reply]
that's how pack.json was writen by me, I took as a basis a sample from the jar file of the version
![Screenshot_20241023_075820](https://github.com/user-attachments/assets/919e951c-8ce1-4cf1-920d-c2694001b8b2)

**Reply by BenCheung0422** [reply]
Then I think the only place of issue is the `font.png` file.

**Reply by Sernimator** [reply]
Thank you, I'll try to fix it

**Reply by Sernimator** [reply]
> Then I think the only place of issue is the `font.png` file.

I tried to replace the texture directly in the game files and everything worked out, but when trying to replace textures with a resource pack, everything goes wrong, maybe I didn't understand something, where can I download other people's resourcepacks and check if they work for me?

**Reply by BenCheung0422** [reply, marked as answer]
Sure. If you want to test specifically the fonts, this is my test pack.
[pack.zip](https://github.com/user-attachments/files/17486067/pack.zip)

**Reply by Sernimator** [reply]
Thank you, your resourcepack is working, it looks like I really missed something. I will try to compare them and fix the errors in my pack

**Reply by Sernimator** [reply]
> Sure. If you want to test specifically the fonts, this is my test pack. [pack.zip](https://github.com/user-attachments/files/17486067/pack.zip)

All this time, the reason was in the zip file.
When I removed all the contents of your zip file and replaced it with my own, the custom graphics worked, but when I created the zip file myself and put the same contents, it doesn't work.
Thank you again for your help
